reana_server/scheduler/consumer.py

```diff
@@ -125,18 +133,50 @@ def get_consumers(self, Consumer, channel):
             )
         ]
 
-    def on_message(self, workflow_submission, message):
+    def _fail_workflow(self, workflow_id: str, logs: str = "") -> None:
+        self.workflow_status_publisher.publish_workflow_status(
+            workflow_id, status=RunStatus.failed.value, logs=logs,
+        )
+
+    def _retry_workflow_submission(
+        self, workflow_id: str, workflow_submission: Dict
+    ) -> None:
+        retry_count = workflow_submission.get("retry_count", 0)
+        if retry_count >= REANA_SCHEDULER_REQUEUE_COUNT:
+            error_message = (
+                f"Workflow {workflow_submission['workflow_id_or_name']} failed to schedule "
+                f"{retry_count + 1} times. Giving up."
```

Comment:
Minor: I guess we don't need to do `retry_count + 1` here? With `REANA_SCHEDULER_REQUEUE_COUNT=10` I got `Workflow 36ea2285-aa10-4664-a2c7-4bf3e1a0e20c failed to schedule 11 times. Giving up.`

Reply:
Another idea will be to completely abandon numbers there and say just "Workflow <id> failed to schedule multiple times. Giving up."

Reply:
Yeah, exactly what I wanted to say, "Workflow failed after 10 retries. Giving up.". sounds much better in my eyes, since 10 is the number which was set by the admin. I also think it's better to print it, just to be sure what is the number used while debugging